### ISSUE TITLE
MoveWithinRange gets blocked by friendly units who shares cell

### ISSUE THREAD

**Issue by forcecore** [OpenRA/OpenRA#13512]
On theory, infantries share cells with friendly units but in practice, they block someone's way who is doing MoveWithinRange

https://youtu.be/MyU3n966C2Y

* If I remove Tanya and retry attack dog leaps as expected
* If I move Tanya when it is stuck in MoveWithinRange state, the unit stays still, stuck in MoveWithinRange

**Comment by GraionDilach** [comment]
Sounds like a duplicate of #5841 to me.
